src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -9,180 +9,387 @@
         const inputLength = debouncedInput.length;
         // Dismiss if match length is greater than 3x the input's length
         if (matchLength > inputLength * 3) return false;
-
         return ({
           title: symbol.title,
           path: symbol.path,
+          parents: this.getParents(symbol.parent),
+          type: symbol.type,
           inputLengthDifference: symbol.title.length - inputLength,
           matchLength,
           matchLengthDifference: matchLength - inputLength,
           start: match.index,
           substring: match[0],
         });
-      }).filter(Boolean)
-    ),
+      }).filter(Boolean);
+    },
+    handleKeyEnter(event) {
+      if (
+        (event.srcElement.id !== 'filter-input' && event.srcElement.id !== MatchId)
+        || this.noResultsWereFound
+      ) return;
+      this.$router.push(this.filteredSymbols[this.focusedIndex].path);
+      this.closeQuickNavigationModal();
+    },
     orderSymbolsByPriority(matchingSymbols) {
       return matchingSymbols.sort((a, b) => {
         // Shortests symbol match title have preference over larger titles
         if (a.matchLengthDifference > b.matchLengthDifference) return 1;
         if (a.matchLengthDifference < b.matchLengthDifference) return -1;
-        // Shortests symbol title have preference over larger titles
-        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
-        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
         // Matches at the beginning of string have relevance over matches at the end
         if (a.start > b.start) return 1;
         if (a.start < b.start) return -1;
+        // Shortests symbol title have preference over larger titles
+        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
+        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
         return 0;
       });
     },
-  },
-  watch: {
-    userInput: 'debounceInput',
+    scrollIntoView() {
+      this.$refs.match[this.focusedIndex].scrollIntoView({
+        block: 'nearest',
+        inline: 'start',
+      });
+    },
+    startingPointHook() {
+      this.focusedIndex = this.totalItemsToNavigate - 1;
+    },
   },
 };
 </script>
 
-<style scoped lang="scss">
+<style lang="scss">
 @import 'docc-render/styles/_core.scss';
 
-  .close-icon {
-    padding-bottom: 10px;
+$modal-margin-top: 10rem;
+$base-border-width: 1px;
+$filter-padding: rem(15px);
+
+.quick-navigation {
+  input[type="text"] {
+    @include font-styles(body-large);
   }
-  .container {
-      flex-direction: row;
+  &__container {
+    background-color: var(--color-fill);
+    border: solid $base-border-width var(--color-fill-gray);
+    border-radius: $border-radius;
+    filter: drop-shadow(0px 7px 50px rgba(0, 0, 0, 0.25));
+    margin: auto;
+    max-width: rem(800px);
+    .filter__wrapper {
+      border: none;
+    }
+    > * {
+      --input-text: var(--color-figure-gray-secondary) !important;
```

Comment:
Have you looked into other ways of making this work without the need for the `!important`?

Reply:
Removed in https://github.com/apple/swift-docc-render/pull/396/commits/62db4a637515e8fd67f83266a028ced48f825d01